buildenv/jenkins/openjdk_tests

```diff
@@ -239,42 +239,93 @@ if (JDK_VERSIONS.size() > 1 || JDK_IMPLS.size() > 1 || PLATFORMS.size() >1 || PL
-                    def SCM_GIT_REPO_VAL = sh(script: "echo ${SCM_GIT_REPO}", returnStdout: true).trim()
-                    SCM_GIT_REPO_VAL = SCM_GIT_REPO_VAL.replace("https://github.com/","git@github.com:")
+                        // SCM_GIT_REPO value only gets expanded in sh
+                        def SCM_GIT_REPO_VAL = sh(script: "echo ${SCM_GIT_REPO}", returnStdout: true).trim()
+                        SCM_GIT_REPO_VAL = SCM_GIT_REPO_VAL.replace("https://github.com/","git@github.com:")
 
-                    sh "git clone -b ${SCM_GIT_BRANCH} ${SCM_GIT_REPO_VAL} openjdk-tests"
-                } else {
-                    def gitConfig = scm.getUserRemoteConfigs().get(0)
+                        sh "git clone -b ${SCM_GIT_BRANCH} ${SCM_GIT_REPO_VAL} openjdk-tests"
+                    } else {
+                        def gitConfig = scm.getUserRemoteConfigs().get(0)
 
-                    // Adopt windows machines require env here https://github.com/AdoptOpenJDK/openjdk-tests/issues/1803
-                    ref_cache = "${env.HOME}/openjdk_cache"
-                    checkout scm: [$class: 'GitSCM',
-                        branches: [[name: "${scm.branches[0].name}"]],
-                        extensions: [
-                            [$class: 'CleanBeforeCheckout'],
-                            [$class: 'CloneOption', reference: ref_cache],
-                            [$class: 'RelativeTargetDirectory', relativeTargetDir: 'openjdk-tests']],
-                        userRemoteConfigs: [[url: "${gitConfig.getUrl()}"]]
-                    ]
+                        // Adopt windows machines require env here https://github.com/AdoptOpenJDK/openjdk-tests/issues/1803
+                        ref_cache = "${env.HOME}/openjdk_cache"
+
+                        checkout scm: [$class: 'GitSCM',
+                            branches: [[name: "${scm.branches[0].name}"]],
+                            extensions: [
+                                [$class: 'CleanBeforeCheckout'],
+                                [$class: 'CloneOption', reference: ref_cache],
+                                [$class: 'RelativeTargetDirectory', relativeTargetDir: 'openjdk-tests']],
+                            userRemoteConfigs: [[url: "${gitConfig.getUrl()}"]]
+                        ]
+                    }
+                    jenkinsfile = load "${WORKSPACE}/openjdk-tests/buildenv/jenkins/JenkinsfileBase"
+                    jenkinsfile.testBuild()
+                } catch (Exception e) {
+                    println("Exception: " + e.toString())
+                    // build result may not be updated correctly at the moment (see https://issues.jenkins.io/browse/JENKINS-56402)
+                    // if there is an exception and build result is not FAILURE, set currentBuild.result to FAILURE
+                    if (currentBuild.currentResult != 'FAILURE' && currentBuild.result != 'FAILURE') {
```

Comment:
Looks like this if condition is to work with the following if condition https://github.com/adoptium/aqa-tests/pull/2530/files#diff-ebf20f64c486ae48bd60b965c555076519f9d5623789f4574467d3051bb530f7R286. 

If the effort is to cover the case result == null probably would be better to try isWorseThan(SUCCESS) https://javadoc.jenkins-ci.org/hudson/model/Result.html#isWorseThan-hudson.model.Result-

Reply:
Yeah, understand that the problem is we can not rely on currentBuild.currentResult or currentBuild.result on the finally clause for notifications or other things as  they aren't set until we get out of the finally. So we need to set currentBuild.result manually. We can set the result if there is exception. My suggestion is could we do the following:

```
...
  } catch(e) {
    currentBuild.result = 'FAILURE'
  } finally {
   ...
    if (currentBuild.result == 'FAILURE')  {// or currentBuild.current == 'FAILURE')  both works
        code for error checking
     }
    ...
  }

Reply:
We do not need to override the result if it is `FAILURE `

Reply:
Talked with @sophia-guo , `currentResult` is set to `SUCCESS` if `result` is null. In other cases, `currentResult` is set to `result`. https://github.com/jenkinsci/workflow-support-plugin/pull/31/files

PR is updated.